scripts/validate.mjs

```diff
@@ -2,40 +2,76 @@ import path from 'path'
 import Ajv from 'ajv'
 import { readdirSync, readFileSync } from 'fs'
 
-const ajv = new Ajv()
-
 const SAMPLES_DIRECTORY = './samples'
 const SCHEMAS_DIRECTORY = './schemas'
 
-forEachFile(SCHEMAS_DIRECTORY, addSchema)
-forEachFile(SAMPLES_DIRECTORY, validateSample)
+validateSchemasObjectsPropertiesCase()
+validateSchemasIds()
+validateSamples()
 
-function addSchema(schemaPath) {
-  const schema = readJson(schemaPath)
+if (process.exitCode !== 0 && process.exitCode !== undefined) {
+  console.log('❌ Some validation errors were found')
+}
 
-  // We need to be careful about schema ids because they need to:
-  // * be unique, or else Ajv will throw
-  // * represent a path, as Ajv will use it to resolve $refs
-  // Here, we make sure that both requirements are respected.
-  const schemaId = computeSchemaIdFromSchemaPath(schemaPath)
-  if (schema.$id !== schemaId) {
-    console.log(`⚠️ Schema ${schemaPath} $id should be ${schemaId}`)
-    process.exitCode = 1
-  }
+function validateSchemasObjectsPropertiesCase() {
+  forEachFile(SCHEMAS_DIRECTORY, (schemaPath) => {
+    const schema = readJson(schemaPath)
 
-  ajv.addSchema(schema)
+    // RUM and telemetry schemas object properties should be snake_case, other schemas objects should
+    // be camelCase
+    const shouldBeSnakeCase =
+      schemaPath.startsWith(`${SCHEMAS_DIRECTORY}/rum/`) || schemaPath.startsWith(`${SCHEMAS_DIRECTORY}/telemetry/`)
+
+    // Some exceptions to the rule above. Ideally they should be fixed in the future.
+    const caseExceptions =
+      schemaPath === `${SCHEMAS_DIRECTORY}/session-replay/common/_common-segment-metadata-schema.json`
+        ? ['records_count', 'index_in_view', 'has_full_snapshot']
+        : schemaPath === `${SCHEMAS_DIRECTORY}/session-replay/common/focus-record-schema.json`
+        ? ['has_focus']
+        : schemaPath === `${SCHEMAS_DIRECTORY}/rum/resource-schema.json`
+        ? ['operationType', 'operationName']
+        : []
+
+    forEachObjectProperty(schema, (key) => {
+      const isCorrectCase = shouldBeSnakeCase ? isSnakeCase(key) : isCamelCase(key)
+      if (!isCorrectCase && !caseExceptions.includes(key)) {
+        console.log(`❌ Schema ${schemaPath} property ${key} is not ${shouldBeSnakeCase ? 'snake_case' : 'camelCase'}`)
```

Comment:
Add clarification if you've looked into another schemas to find the correct casing.

```suggestion
        console.log(`❌ Schema ${schemaPath} property ${key} is not ${shouldBeSnakeCase ? 'snake_case' : 'camelCase'}`)
        console.log("RUM and Telemetry schemas should be snake_case, other schemas should be camelCase.")
```